relayer/workers/parachaincommitmentrelayer/parachaincommitment/catchup.go

```diff
@@ -196,3 +200,84 @@ func (li *Listener) checkIfDigestItemContainsNonce(
 	}
 	return false, nil
 }
+
+func (li *Listener) checkIncentivizedMessageNonces(
+	digestItem *chainTypes.AuxiliaryDigestItem,
+	nonceToFind uint64,
+) (bool, error) {
+	messages, err := li.getIncentivizedMessages(digestItem.AsCommitment.Hash)
+	if err != nil {
+		return false, err
+	}
+
+	for _, message := range messages {
+		if message.Nonce <= nonceToFind {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func (li *Listener) getBasicMessages(commitment types.H256) ([]chainTypes.BasicOutboundChannelMessage, error) {
+	storageKey, err := parachain.MakeStorageKey(substrate.ChannelID{IsBasic: true}, commitment)
+	if err != nil {
+		return nil, err
+	}
+
+	data, err := li.parachainConnection.GetAPI().RPC.Offchain.LocalStorageGet(rpcOffchain.Persistent, storageKey)
+	if err != nil {
+		li.log.WithError(err).Error("Failed to read commitment from offchain storage")
+		return nil, err
+	}
+
+	if data != nil {
+		li.log.WithFields(logrus.Fields{
+			"commitmentSizeBytes": len(*data),
+		}).Debug("Retrieved commitment from offchain storage")
+	} else {
+		li.log.WithError(err).Error("Commitment not found in offchain storage")
+		return nil, err
+	}
+
+	var messages []chainTypes.BasicOutboundChannelMessage
+
+	err = types.DecodeFromBytes(*data, &messages)
+	if err != nil {
+		li.log.WithError(err).Error("Failed to decode commitment messages")
+		return nil, err
+	}
+
+	return messages, nil
+}
+
+func (li *Listener) getIncentivizedMessages(commitment types.H256) ([]chainTypes.IncentivizedOutboundChannelMessage, error) {
+	storageKey, err := parachain.MakeStorageKey(substrate.ChannelID{IsBasic: true}, commitment)
```

Comment:
This should be `IsIncentivized: true`, right?

Reply:
fixed